MetaMask/metamask-extension
# Initial changes to support adding multiple nfts in one user interaction

## Explanation

Relevant issue here: https://github.com/MetaMask/metamask-extension/issues/17696

Currently you can only add a single NFT of a single contract to your MM wallet in one user interaction.  The user currently needs to repeat this flow for as many NFTs as they have from a given contract.

This change proposes a user flow to add multiple NFTs from the same contract to your MM wallet in one user interaction as opposed to N interactions for N NFTs.

On an implementation level, I created two new actions that handle multiple token Ids at once in batch.  One action checks for ownership and the second action checks for the token details of the NFTs the user intends to add.  These are both very similar to the actions that exist that handle single NFT additions, these could potentially be merged such that one function handles both cases.

On the frontend, a new checkbox has been added that toggles settings between adding one NFT or multiple NFTs in one user interaction.  I'm looking for suggestions and improvements for this UX, I'm not certain if there's a better component to use (ie. a dropdown, radio group, etc) or if the input schema should change.  Currently, the form accepts a string of comma separated NFT Ids.

## Screenshots/Screencaps


### Before (multiple executions of add NFT user flow)

![oldMultiAddNFT](https://user-images.githubusercontent.com/33137497/231623593-56b9324f-d19a-4d87-a135-befee8b847da.gif)

### After

![addMultipleNFTs](https://user-images.githubusercontent.com/33137497/231622431-600ede7e-a12c-46b5-9e5e-e5324b4adfa5.gif)

## Manual Testing Steps

1. Navigate to the NFT tab
2. Click the checkbox under the `Add multiple NFT` title
3. Enter NFT contract address
4. Enter a comma-separated list of NFT ids into the form ie. `1,2`
5. Submit and see multiple NFTs added in one interaction

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
